src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.Prerendering.cs

```diff
@@ -229,9 +229,16 @@ async Task Execute()
                 // Clear all pending work.
                 _nonStreamingPendingTasks.Clear();
 
-                // new work might be added before we check again as a result of waiting for all
-                // the child components to finish executing SetParametersAsync
-                await pendingWork;
+                try
+                {
+                    // new work might be added before we check again as a result of waiting for all
+                    // the child components to finish executing SetParametersAsync
+                    await pendingWork;
+                }
+                catch (NavigationException navigationException)
+                {
+                    await HandleNavigationException(_httpContext, navigationException);
+                }
```

Comment:
Is this specific to the EndpointHtmlRenderer? Or could it happen in one of the interactive modes

Reply:
Interactive is not throwing `NavigationException` to redirect. I don't know how it could happen there.